stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -122,374 +107,190 @@ internal final class _DataStorage {
     @usableFromInline
     var _deallocator: ((UnsafeMutableRawPointer, Int) -> Void)?
     @usableFromInline
-    var _backing: Backing = .swift
-    @usableFromInline
     var _offset: Int
     
     @usableFromInline
     var bytes: UnsafeRawPointer? {
-        @inlinable
-        get {
-            switch _backing {
-            case .swift:
-                return UnsafeRawPointer(_bytes)?.advanced(by: -_offset)
-            case .immutable:
-                return UnsafeRawPointer(_bytes)?.advanced(by: -_offset)
-            case .mutable:
-                return UnsafeRawPointer(_bytes)?.advanced(by: -_offset)
-            case .customReference(let d):
-                return d.bytes.advanced(by: -_offset)
-            case .customMutableReference(let d):
-                return d.bytes.advanced(by: -_offset)
-            @unknown default:
-                fatalError("Unknown Data backing type")
-            }
-        }
+        return UnsafeRawPointer(_bytes)?.advanced(by: -_offset)
     }
 
     @usableFromInline
```

Comment:
This should be @inlinable.
Otherwise (at least currently) the performance of client functions is pretty bad because they have to allocate closure contexts.
Making this inlinable will still keep the interface to _DataStorage resilient. All used member in this function are resilient (@usableFromInline).

Reply:
I will go over the _DataStorage methods this afternoon and mark them accordingly. Thanks for the extra pair of eyes.